blogfrontend/src/components/Signup.js

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+
+const Signup = () => {
+  const [username, setUserName] = useState({
+    username: "",
+    email: "",
+    password: "",
+    cpassword: "",
+  });
+
+  // if(username===""){
+  //  alert("enter name")
+  // }
+
+  const handlChange = (e) => {
+    const { name, value } = e.target;
+    console.log(name, value);
+    setUserName({
+      ...username,
+      [name]: value,
+    });
+  };
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log("datZ", e.target.name);
+    if (!username.username) {
+      alert("Enter your  username");
+    } else if (!username.email) {
```

Comment:
you have not validated the email. Like you need to check it should contain @ . etc.